data-serving/data-service/src/controllers/case.ts

```diff
@@ -357,20 +357,40 @@ export const upsert = async (req: Request, res: Response): Promise<void> => {
  * Handles HTTP DELETE /api/cases.
  */
 export const batchDel = async (req: Request, res: Response): Promise<void> => {
-    Case.deleteMany(
-        {
-            _id: {
-                $in: req.body.caseIds,
+    if (req.body.caseIds !== undefined) {
+        Case.deleteMany(
+            {
+                _id: {
+                    $in: req.body.caseIds,
+                },
             },
-        },
-        (err) => {
-            if (err) {
-                res.status(500).json(err.message);
-                return;
-            }
-            res.status(204).end();
-        },
-    );
+            (err) => {
+                if (err) {
+                    res.status(500).json(err.message);
+                    return;
+                }
+                res.status(204).end();
+            },
+        );
+        return;
+    }
+    if (req.body.query.trim() === '') {
+        res.status(422).json('query must not be empty');
```

Comment:
not sure if we can do this as part of the openapi spec?

Reply:
Ah, you can specify a regex, done.